paimon-core/src/main/java/org/apache/paimon/io/KeyValueDataFileWriter.java

```diff
@@ -177,6 +179,7 @@ public DataFileMeta result() throws IOException {
                         ? DataFileIndexWriter.EMPTY_RESULT
                         : dataFileIndexWriter.result();
 
+        String externalPath = isExternalPath ? path.getParent().toString() : null;
```

Comment:
When isExternalPath is true?

Reply:
have been removed, will be introduced in next PR.